src/Microsoft.AspNetCore.Mvc.Razor/TagHelpers/FeatureTagHelperTypeResolver.cs

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.AspNetCore.Razor;
+using Microsoft.AspNetCore.Mvc.ApplicationParts;
+using Microsoft.AspNetCore.Razor.Runtime.TagHelpers;
+using System.Reflection;
+
+namespace Microsoft.AspNetCore.Mvc.Razor.TagHelpers
+{
+    /// <summary>
+    /// Resolves tag helper types from the <see cref="ApplicationPartManager.ApplicationParts"/>
+    /// of the application.
+    /// </summary>
+    public class FeatureTagHelperTypeResolver : ITagHelperTypeResolver
+    {
+        private readonly ApplicationPartManager _manager;
+
+        /// <summary>
+        /// Initializes a new <see cref="FeatureTagHelperTypeResolver"/> instance.
+        /// </summary>
+        /// <param name="manager">The <see cref="ApplicationPartManager"/> of the application.</param>
+        public FeatureTagHelperTypeResolver(ApplicationPartManager manager)
+        {
+            if (manager == null)
+            {
+                throw new ArgumentNullException(nameof(manager));
+            }
+
+            _manager = manager;
+
+        }
+
+        /// <inheritdoc />
+        public IEnumerable<Type> Resolve(string assemblyName, SourceLocation documentLocation, ErrorSink errorSink)
```

Comment:
Any reason you can't just override the `GetExportedTypes` method? That way you would not need to have all the try catch error logic (could delegate to the base class)